test-common/src/main/java/org/apache/kafka/common/test/TestUtils.java

```diff
@@ -125,4 +148,131 @@ static void retryOnExceptionWithTimeout(final Runnable runnable) throws Interrup
             Thread.sleep(DEFAULT_POLL_INTERVAL_MS);
         }
     }
+
+    /**
+     * Wrap a single record log buffer.
+     */
+    public static MemoryRecords singletonRecords(byte[] value,
+                                                 byte[] key,
+                                                 Compression codec,
+                                                 long timestamp,
+                                                 byte magicValue) {
+        return records(Collections.singletonList(new SimpleRecord(timestamp, key, value)), magicValue, codec);
+    }
+
+    public static MemoryRecords singletonRecords(byte[] value, byte[] key) {
```

Comment:
this method is equal to `MemoryRecords.withRecords(Compression.NONE, new SimpleRecord(key, value));`, so we don't need to add this helper, right?